### PR TITLE
#164139638 Setup react router version 4 (react-router-dom)

### PR DESCRIPTION
**Pull request by kleva-j**
#### What does this PR do?
Sets up routing functionality for the application.

#### Description of Task to be completed?
- install and use react-router-dom for routing.
- create the folder structure for the components and pages.
- create the home, log in and 404 components.
- enable routing to the homepage, login page and 404 pages.

#### How should this be manually tested?
- run `npm run start:dev`, this will launch the app and would load up the landing page in your default browser.
- navigate to this route `http://localhost:3000/login` to see the login page.
- navigate to a route that does not exist and you would get a 404 page.

#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
#164139638

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "main": "index.js",
   "scripts": {
     "webpack": "webpack",
+    "build": " ",
```
Comment (codinger41) [inline package.json]: @kleva-j  Hi Michael, what does this do?

Comment (kleva-j) [reply package.json]: It was a fix for an error that prevented me from installing packages as there was no build script.

**Review by youngestdj — DISMISSED** [review]

**Review by codinger41 — DISMISSED** [review]

**Comment by 9jaswag** [comment]
@kleva-j there are conflicts to be fixed here.

**Review by olajide1234 — DISMISSED** [review]

**Review by youngestdj — DISMISSED** [review]

File: src/app.js [hunk 2]
```diff
@@ -1,22 +1,12 @@
 import React from 'react';
-import ReactDOM from 'react-dom';
-import { Provider } from 'react-redux';
-import store from './redux/store';
+import { BrowserRouter as Router } from 'react-router-dom';
+import Routes from './route/index';
```
Comment (9jaswag) [inline src/app.js]: @kleva-j you don't need to add the `/index` part. I mentioned this yesterday. If you want to reference an `Index.js/jsx` file in a directory, just referencing the directory name will fetch the index file.

File: src/route/index.jsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import { Switch, Route } from 'react-router-dom';
+import Home from '../components/Home';
+import NotFound from '../components/404';
+import Login from '../components/auth/Login';
+
+const Routes = () => (
```
Comment (9jaswag) [inline src/route/index.jsx]: @kleva-j this file should be named `Index.jsx` as it is a component.

**Review by 9jaswag — CHANGES_REQUESTED** [review]

File: src/app.js (hunk 2 above)
Comment (kleva-j) [reply src/app.js]: I actually did that and it worked but eslint was flagging it as an error. It said `Casing of ./route does not match the underlying filesystem.eslint(import/no-unresolved)`.

Comment (kleva-j) [reply src/app.js]: I have made updates, but the tests are failing on Circle Ci because of the missing `/Index
` part.

**Review by codinger41 — APPROVED** [review]

**Review by youngestdj — APPROVED** [review]

**Review by 9jaswag — APPROVED** [review]

**Comment by 9jaswag** [comment]
@kleva-j you missed the feedback on renaming the PR title

**Pull request merged**